packages/next/src/server/lib/cache-handlers/default.ts

```diff
@@ -0,0 +1,47 @@
+/*
+  This is the default "use cache" handler it defaults
+  to an in memory store
+*/
+import type { CacheEntry, CacheHandler } from './types'
+
+interface DefaultCacheEntry extends CacheEntry {
+  // For the default cache we store errored cache
+  // entries and allow them to be used up to 3 times
+  // after that we want to dispose it and try for fresh
+
+  // If an entry is errored we return no entry
+  // three times so that we retry hitting origin (MISS)
+  // and then if it still fails to set after the third we
+  // return the errored content and use expiration of
+  // Math.min(30, entry.expiration)
+  isErrored: boolean
+  errorRetryCount: number
+
+  // compute size on set since we need to read size
+  // of the ReadableStream for LRU evicting
+  size?: number
+}
+
+export class DefaultCacheHandler implements CacheHandler {
```

Comment:
Let's use a singleton instead of classes. No need to instantiate new ones all the time.